src/lib/finishPollMessage.ts

```diff
@@ -78,11 +78,13 @@ export async function finishPollMessage({
             .getEnvironmentReader()
             .getSettings()
             .getById('wordcloud-api');
-        const timeZone = await read
-            .getEnvironmentReader()
-            .getSettings()
-            .getById('timezone');
 
+        const user = await read.getUserReader().getById(data.user.id);
+        const utcOffset = await user.utcOffset;
+        const usertime = await timeZones.timeZones.find(time => time.offset === utcOffset) as any;
+        const timeZone = await read.getEnvironmentReader().getSettings().getById('timezone');
+        const timezone = usertime ? usertime.utc[0]:timeZone; 
```

Comment:
Let's create a helper function here to resolve the timezone info

Reply:
Sure ☺👍